corehq/apps/geospatial/views.py

```diff
@@ -435,3 +436,16 @@ def get_users_with_gps(request, domain):
     ]
 
     return json_response({'user_data': user_data})
+
+
+@method_decorator(toggles.GEOSPATIAL.required_decorator(), name="dispatch")
+class CasesReassignmentView(BaseDomainView):
+    urlname = "reassign_cases"
+
+    def post(self, request, domain, *args, **kwargs):
+        request_data = json.loads(request.body)
+        case_id_to_owner_id = request_data.get('case_id_to_owner_id', {})
+
+        update_cases_owner(domain, case_id_to_owner_id)
```

Comment:
Given that this API will be consumed from only a single known source (Case Management Page in this case) and it is expected to have the valid case ids, I don't think it is worth checking for this use case

Reply:
Hey @ajeety4 

Valid point about the function only being used internally. The view however is open to receiving data from anywhere via the URL, so its not safe to assume that it can only be called via expected means.

As for the ids being valid, I believe such a check should be at a common point like in CaseBlock and it does seem that it would fail when creating the XML if IDs are in bad format.
For valid IDs, I assume the form submission would fail if the case ID is not of the domain itself. Good to confirm that just for sanity. May be add a test.

Reply:
Echoing Manish's point that there is POST endpoint for passing in a JSON data, so there is a possibility of someone intentionally passing in bad data.
 
I think a simple check of doing a search of all the caseids on case-search index and userids on users index should work here. Let me know if you need any assistance with it.

Reply:
The check for valid case ids and owner ids were added in c7070fec1d4009193b698ec5bc8aad4065d427a0 and 353dec76667bd5b8dddd7c7798bc95f558633959 respectively